### ISSUE TITLE
Validation fail with DL3061 error when Dockerfile begin with comment that start with 'syntax'  

### ISSUE THREAD

**Issue by StefanoGuerrini** [hadolint/hadolint#795]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behaviour
Success

### Actual behaviour
DL3061 error: Invalid instruction order. Dockerfile must begin with `FROM`, `ARG` or comment.

### Steps to reproduce the behaviour
A Dockerfile that begin with a comment that starts with 'syntax'

Run:
`docker run --rm -i hadolint/hadolint:latest hadolint "$@" - < Dockerfile`

Version:
  `Haskell Dockerfile Linter v2.9.1-0-g42d0378-dirty`

Error:
```bash
-:1 DL3061 error: Invalid instruction order. Dockerfile must begin with `FROM`, `ARG` or comment.
```

**Dockerfile** 
```Dockerfile
# syntax=something
ARG FOO=BAR
FROM debian:8
```

**Additional environment details (OS, stack version, etc.)**
Using `hadolint/hadolint:2.8.0-debian` works

**Comment by zeeveener** [comment]
This affects all the `latest-X` and `v2.9.1-X` versions. The solution, as mentioned in the description, is to pin the version to `v2.8.0-X`
